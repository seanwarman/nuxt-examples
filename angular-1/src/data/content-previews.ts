import { RoomPlannerImages, SectionImages } from '../assets/images';

export interface ContentPreview {
  label: string;
  primaryImage: string;
  icon: string;
  body: string;
  slug: string;
  firstSubsection?: string;
  learnIcon: string;
  designIcon: string;
}

export const contentPreviews: ContentPreview[] = [
  {
    slug: 'camera-and-audio',
    label: 'Camera & Audio',
    firstSubsection: 'camera',
    primaryImage: SectionImages.cameraAndAudio.contentPreviewHero,
    icon: SectionImages.cameraAndAudio.icons.sectionIcon,
    designIcon: SectionImages.cameraAndAudio.icons.designIcon,
    learnIcon: SectionImages.cameraAndAudio.icons.learnIcon,
    body: `We've tuned and tweaked the hardware. Now, position it correctly to
     unlock the best performance.`,
  },
  {
    slug: 'displays-and-extras',
    label: 'Displays & Extras',
    firstSubsection: 'screen',
    primaryImage: SectionImages.displaysAndExtras.contentPreviewHero,
    icon: SectionImages.displaysAndExtras.icons.sectionIcon,
    designIcon: SectionImages.displaysAndExtras.icons.designIcon,
    learnIcon: SectionImages.displaysAndExtras.icons.learnIcon,
    body: `Correctly sizing displays is a key part of the conference experience.
      Plus, get some hints and tips on using the Series One Board 65 and Desk 27,
      developed in collaboration with Avocor,
      in your space.`,
  },
  {
    slug: 'room-layout',
    label: 'Room layout',
    firstSubsection: 'viewing-area',
    primaryImage: SectionImages.roomLayout.contentPreviewHero,
    icon: SectionImages.roomLayout.icons.sectionIcon,
    designIcon: SectionImages.roomLayout.icons.designIcon,
    learnIcon: SectionImages.roomLayout.icons.learnIcon,
    body: `Following a few spatial guidelines can make a huge difference for remote and in-room participants.`,
  },
  {
    slug: 'furniture-lighting',
    label: 'Furniture and lighting',
    firstSubsection: 'lighting',
    primaryImage: SectionImages.furnitureLighting.contentPreviewHero,
    icon: SectionImages.furnitureLighting.icons.sectionIcon,
    designIcon: SectionImages.furnitureLighting.icons.designIcon,
    learnIcon: SectionImages.furnitureLighting.icons.learnIcon,
    body: `Make your meeting room look and feel inviting with simple decorative and lighting tips. Appropriate lighting helps both local and remote participants have a great experience.`,
  },
  {
    slug: 'acoustics',
    label: 'Acoustics',
    firstSubsection: 'separation',
    primaryImage: SectionImages.acoustics.contentPreviewHero,
    icon: SectionImages.acoustics.icons.sectionIcon,
    designIcon: SectionImages.acoustics.icons.designIcon,
    learnIcon: SectionImages.acoustics.icons.learnIcon,
    body: `Poor acoustics is a common downfall of video conferencing spaces. The right allowance of acoustic finishes can help make speech clear and audible, and adequate isolation helps avoid interruptions.`,
  },
  {
    slug: 'services',
    label: 'Services',
    firstSubsection: 'air-handling',
    primaryImage: SectionImages.services.contentPreviewHero,
    icon: SectionImages.services.icons.sectionIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Make sure power is provisioned for the right places, and ensure ventilation systems don't cause unwanted noise pickup on your microphones.`,
  },
  {
    slug: 'room-planner',
    label: 'Room Designer',
    primaryImage: RoomPlannerImages.contentPreviewHero,
    icon: RoomPlannerImages.icons.mainIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Need to size up your room? All the best practices in this guide are used to help you pick the equipment and recommended room size for the occupancy you need.`,
  },
];
