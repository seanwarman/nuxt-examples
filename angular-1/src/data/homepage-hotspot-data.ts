export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  cameraPosition: {
    x: number;
    y: number;
    z: number;
  };
  cameraRotation: {
    x: number;
    y: number;
    z: number;
  };
  modelsOn: Array<string>;
  modelsOff: Array<string>;
}> = [
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -5,
    y: 113,
    z: -160,
    cameraPosition: {
      x: -100,
      y: 175,
      z: 350,
    },
    cameraRotation: {
      x: 100,
      y: 100,
      z: 0,
    },
    modelsOn: ['CameraAndAudio', 'defaultChairs', 'acousticDefault'],
    modelsOff: ['defaultCameraAndAudio'],
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -230,
    y: 150,
    z: 50,
    cameraPosition: {
      x: 100,
      y: 200,
      z: 500,
    },
    cameraRotation: {
      x: 50,
      y: 100,
      z: 0,
    },
    modelsOn: [
      'DisplaysAndExtras',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: 90,
    y: 20,
    z: 200,
    cameraPosition: {
      x: 0,
      y: 280,
      z: 800,
    },
    cameraRotation: {
      x: 150,
      y: 100,
      z: 100,
    },
    modelsOn: [
      'RoomLayout',
      'defaultChairs',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: 110,
    y: 70,
    z: 80,
    cameraPosition: {
      x: 400,
      y: 200,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 70,
      z: -100,
    },
    modelsOn: [
      'FurnitureAndLighting',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['defaultChairs'],
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -230,
    y: 200,
    z: 180,
    cameraPosition: {
      x: 300,
      y: 800,
      z: -150,
    },
    cameraRotation: {
      x: -150,
      y: 0,
      z: -100,
    },
    modelsOn: [
      'Acoustics',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticPanel00',
      'acousticPanel01',
      'acousticPanel02',
      'acousticPanel03',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['stripLight00', 'stripLight01', 'acousticDefault'],
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 110,
    y: 115,
    z: -160,
    cameraPosition: {
      x: 300,
      y: 160,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 50,
      z: -250,
    },
    modelsOn: [
      'services',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-planner',
    text: 'Room Designer',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-designer-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    cameraPosition: {
      x: -30,
      y: 150,
      z: 150,
    },
    cameraRotation: {
      x: 40,
      y: 60,
      z: 0,
    },
    modelsOn: [
      'Drawing',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
];
