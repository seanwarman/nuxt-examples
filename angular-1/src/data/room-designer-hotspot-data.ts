export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  camX: number;
  camY: number;
  camZ: number;
}> = [
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: -110,
    y: 20,
    z: 100,
    camX: -800,
    camY: 800,
    camZ: 800,
  },
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    camX: -800,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: 0,
    y: 110,
    z: -160,
    camX: -1200,
    camY: 500,
    camZ: 600,
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 120,
    y: 200,
    z: -160,
    camX: -120,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: -70,
    y: 265,
    z: 20,
    camX: -120,
    camY: 300,
    camZ: 400,
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -210,
    y: 190,
    z: -160,
    camX: -120,
    camY: 220,
    camZ: 580,
  },
  // {
  //   name: 'room-planner',
  //   text: 'Room Planner',
  //   color: '#0d9cab',
  //   defaultIcon: 'icon-hotspot-teal-01@3x.png',
  //   activeIcon: 'icon-room-designer-white-teal-circle-01@3x.png',
  //   halo: 'teal.png',
  //   x: -10,
  //   y: 150,
  //   z: -160,
  //   camX: 0,
  //   camY: 200,
  //   camZ: 200,
  // },
];
