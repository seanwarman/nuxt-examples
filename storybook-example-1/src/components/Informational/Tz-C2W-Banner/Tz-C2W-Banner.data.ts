export const exampleData = {
  bannerImage: {
    src: '/lifestyle.jpeg',
    alt: 'lifestyle Image'
  },
  bannerQuote: '"I didnt know I could claim a mountain bike with a cycle to work scheme!"',
  slogan: 'Ride roads on the weekdays, shred trails on the weekends.',
  label: 'Apply Now',
  link: {
    url: 'https://google.com/',
    target: '_blank'
  }
};
