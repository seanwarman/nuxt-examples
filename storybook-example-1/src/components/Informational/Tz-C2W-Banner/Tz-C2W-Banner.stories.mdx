import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';

import TzC2WBanner from './Tz-C2W-Banner.vue';

<Meta
  title="components/Informational/C2W Banner"
  component={TzC2WBanner}
  argTypes={{
    isResponsive: {
      type: { name: 'boolean' }
    }
  }}
/>

export const Template = (args) => ({
  components: { TzC2WBanner },
  setup() {
    return { args };
  },
  template: '<TzC2WBanner v-bind="args"/>'
});

# C2W Banner

<Canvas withSource="open">
  <Story
    name="Default"
    args={{
      bannerImage: {
        src: '/lifestyle.jpeg',
        alt: 'lifestyle Image'
      },
      isResponsive: true
    }}
    parameters={{
      // Set the viewports in Chromatic at a component level.
      chromatic: { viewports: [768, 1024, 1300, 1500] }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Default" />

## Not Responsive

<Canvas withSource="open">
  <Story
    name="Not Responsive"
    args={{
      bannerImage: {
        src: '/lifestyle.jpeg',
        alt: 'lifestyle Image'
      },
      isResponsive: false
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
