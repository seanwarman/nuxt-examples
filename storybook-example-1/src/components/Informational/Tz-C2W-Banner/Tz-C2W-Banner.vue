<template>
  <article
    class="relative flex h-[520px] flex-col justify-between px-3 py-5 lg:h-[710px]"
    :class="{
      'sm:static sm:grid sm:grid-cols-3 sm:grid-rows-6 sm:justify-start sm:gap-5 sm:bg-transparent md:grid-cols-4 ':
        isResponsive
    }"
  >
    <div class="relative z-10 pt-1" :class="{ 'sm:col-span-1 sm:col-start-1 ': isResponsive }">
      <h2
        class="text-center text-xs font-bold uppercase text-white"
        :class="{
          'sm:text-left sm:text-sm sm:tracking-wider sm:text-dark-fade': isResponsive
        }"
      >
        <slot name="title"> cycle to work </slot>
      </h2>
    </div>
    <div class="relative z-10 hidden" :class="{ 'sm:row-span-2 sm:block': isResponsive }">
      <h1 class="max-w-[140px] text-xl font-bold">
        <slot name="heading">Get your bike on the C2W scheme!</slot>
      </h1>
    </div>
    <div
      class="flex flex-col items-center gap-2"
      :class="{
        'sm:relative sm:col-span-2 sm:col-start-2 sm:row-span-full sm:row-start-1 sm:items-start sm:justify-end sm:p-5 md:col-span-3 md:col-start-2 md:p-12 lg:p-16':
          isResponsive
      }"
    >
      <p
        class="relative z-10 m-0 max-w-xs text-center text-3xl font-black leading-9 text-white"
        :class="{ 'sm:text-left sm:text-2xl ': isResponsive }"
      >
        <slot name="quote">I didn't know I could claim a mountain bike with a cycle to work scheme!"</slot>
      </p>
      <p
        class="relative z-10 m-0 hidden text-sm font-semibold text-white"
        :class="{ 'sm:inline md:text-lg': isResponsive }"
      >
        <slot name="slogan">Ride roads on the weekdays and shred trails on the weekend.</slot>
      </p>
      <img class="absolute top-0 left-0 z-0 h-full w-full object-cover brightness-75" :src="bannerImage.src" />
    </div>
    <div class="relative z-10">
      <tz-button :link="link" class="w-full" :class="{ 'sm:w-auto': isResponsive }"
        ><slot name="label">apply now</slot>
      </tz-button>
    </div>
  </article>
</template>

<script setup lang="ts">
import { toRefs } from 'vue';
import TzButton from '@/components/Action/Tz-Button/Tz-Button.vue';
import type { Image, Link } from '@/components/Informational/types';

export interface Props {
  bannerImage: Image;
  isResponsive?: boolean;
  link: Link;
}

const props = withDefaults(defineProps<Props>(), {
  isResponsive: true
});

const { bannerImage, isResponsive } = toRefs(props);
</script>
