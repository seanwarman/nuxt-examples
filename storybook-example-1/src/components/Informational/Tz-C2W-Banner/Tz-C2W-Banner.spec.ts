/**
 * @vitest-environment jsdom
 */

import { mount } from '@vue/test-utils';
import TzC2WBanner from './Tz-C2W-Banner.vue';
import { expect, test } from 'vitest';
import { exampleData } from './Tz-C2W-Banner.data';

const wrapper = mount(TzC2WBanner, {
  props: { ...exampleData }
});

test('renders correctly', () => {
  expect(wrapper.element).toMatchSnapshot();
});
