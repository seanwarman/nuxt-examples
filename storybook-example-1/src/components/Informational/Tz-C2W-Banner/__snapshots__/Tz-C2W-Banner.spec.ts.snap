// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<article
  bannerquote="\\"I didnt know I could claim a mountain bike with a cycle to work scheme!\\""
  class="relative flex h-[520px] flex-col justify-between px-3 py-5 lg:h-[710px] sm:static sm:grid sm:grid-cols-3 sm:grid-rows-6 sm:justify-start sm:gap-5 sm:bg-transparent md:grid-cols-4"
  label="Apply Now"
  slogan="Ride roads on the weekdays, shred trails on the weekends."
>
  <div
    class="relative z-10 pt-1 sm:col-span-1 sm:col-start-1"
  >
    <h2
      class="text-center text-xs font-bold uppercase text-white sm:text-left sm:text-sm sm:tracking-wider sm:text-dark-fade"
    >
      
       cycle to work 
      
    </h2>
  </div>
  <div
    class="relative z-10 hidden sm:row-span-2 sm:block"
  >
    <h1
      class="max-w-[140px] text-xl font-bold"
    >
      
      Get your bike on the C2W scheme!
      
    </h1>
  </div>
  <div
    class="flex flex-col items-center gap-2 sm:relative sm:col-span-2 sm:col-start-2 sm:row-span-full sm:row-start-1 sm:items-start sm:justify-end sm:p-5 md:col-span-3 md:col-start-2 md:p-12 lg:p-16"
  >
    <p
      class="relative z-10 m-0 max-w-xs text-center text-3xl font-black leading-9 text-white sm:text-left sm:text-2xl"
    >
      
      I didn't know I could claim a mountain bike with a cycle to work scheme!"
      
    </p>
    <p
      class="relative z-10 m-0 hidden text-sm font-semibold text-white sm:inline md:text-lg"
    >
      
      Ride roads on the weekdays and shred trails on the weekend.
      
    </p>
    <img
      class="absolute top-0 left-0 z-0 h-full w-full object-cover brightness-75"
      src="/lifestyle.jpeg"
    />
  </div>
  <div
    class="relative z-10"
  >
    <a
      class="flex items-center justify-center gap-2 text-sm font-extrabold uppercase tracking-wider focus:ring bg-dark hover:bg-black text-white ring-dark px-8 py-3 rounded-[3px] w-full sm:w-auto"
      href="https://google.com/"
      target="_blank"
    >
      
      
      
      
      apply now
      
      
    </a>
  </div>
</article>
`;
